odoo/addons/base/tests/test_res_country.py

```diff
@@ -1,21 +1,63 @@
 from odoo.tests import TransactionCase, tagged
 
 
-@tagged('-at_install', 'post_install')
-class TestResCountryState(TransactionCase):
-    def test_find_by_name(self):
-        """It should be possible to find a state by its display name
-        """
-        glorious_arstotzka = self.env['res.country'].create({
+class TestResCountryCommon(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.glorious_arstotzka = cls.env['res.country'].create({
             'name': 'Arstotzka',
             'code': 'AA',
         })
-        altan = self.env['res.country.state'].create({
-            'country_id': glorious_arstotzka.id,
+        cls.altan = cls.env['res.country.state'].create({
+            'country_id': cls.glorious_arstotzka.id,
             'code': 'AL',
             'name': 'Altan',
         })
 
+
+@tagged('-at_install', 'post_install')
+class TestResCountry(TestResCountryCommon):
+    def test_name_search1(self):
+        glorious_arstotzka_tuple = (self.glorious_arstotzka.id, self.glorious_arstotzka.display_name)
+        for (name, op) in [
+            ('ARSTOTZKA', '='),
+            ('arstotzka', '='),
+            ('ARSTOTZKA', '!='),
+            ('arstotzka', '!='),
+            (['ARSTOTZKA'], 'in'),
+            (['arstotzka'], 'in'),
+            (['ARSTOTZKA'], 'not in'),
+            (['arstotzka'], 'not in'),
+        ]:
+            with self.subTest((name, op)):
+                if op in ('!=', 'not in'):
+                    self.assertFalse(
+                        glorious_arstotzka_tuple in
+                        self.env['res.country'].name_search(name, operator=op),
+                        f"Failed on: '{name} {op}'",
+                    )
+                else:
+                    self.assertEqual(
+                        self.env['res.country'].name_search(name, operator=op),
+                        [glorious_arstotzka_tuple],
+                        f"Failed on operator: '{name} {op}'",
+                    )
+
+    def test_name_search2(self):
```

Comment:
NIT: try not using xxx_1, xxx_2 if possible
Maybe `test_name_search` and `test_name_search_code`